src/Microsoft.Extensions.Logging.Analyzers/UseCompiledLogMessagesAnalyzer.cs

```diff
@@ -0,0 +1,55 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Extensions.Logging.Internal;
+
+namespace Microsoft.Extensions.Logging.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class UseCompiledLogMessagesAnalyzer : DiagnosticAnalyzer
```

Comment:
Can we stick this logic into existing analyzer?

Reply:
I guess we could